components/tidb_query/src/codec/mysql/duration.rs

```diff
@@ -695,7 +680,32 @@ impl Duration {
             0,
             fsp,
         );
-        Ok(dur)
+        (Some(dur), None)
+    }
+
+    pub fn from_i64(ctx: &mut EvalContext, n: i64, fsp: u8) -> Result<Duration> {
+        let (dur, err) = Duration::from_i64_without_ctx(n, fsp);
+        match err {
+            Some(e) => {
+                if e.is_overflow() {
+                    ctx.handle_overflow_err(e)?;
+                    if let Some(d) = dur {
+                        Ok(d)
+                    } else {
+                        Err(box_err!("Expect a not none result here, this is a bug"))
```

Comment:
Should we use `unreachable!` here?

Reply:
https://github.com/tikv/tikv/pull/5423#discussion_r323102836

> However, the parse function is very likely to has some small bug, but such bug is not fatal, so if we unwrap here, the tikv will panic because of such small bug, which is very bad for our user